### PR TITLE
add clip package

### PR DESCRIPTION
**Pull request by paul-reilly**
Adds clip:

https://github.com/dacap/clip

"Library to copy/retrieve content to/from the clipboard/pasteboard."

### CODE REVIEW

**Comment by paul-reilly** [comment]
It's building everywhere but Windows shared targets. They are failing on an unresolved external from `check_cxxsnippets` test:
```bash
error LNK2019: unresolved external symbol "bool __cdecl clip::set_text(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" 
```
 I tried moving the dll from `bin` to `lib` but same result.

File: packages/c/clip/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DCLIP_EXAMPLES=OFF")
+        table.insert(configs, "-DCLIP_TESTS=OFF")
+        import("package.tools.cmake").install(package, configs, { buildir="build"})
+        os.cp("clip.h", package:installdir("include/clip"))
+        os.trycp("build/*.a", package:installdir("lib"))
+        os.trycp("build/*.dylib", package:installdir("lib"))
+        os.trycp("build/*.so", package:installdir("lib"))
+        os.trycp("build/*.lib", package:installdir("lib"))
+        os.trycp("build/*.dll", package:installdir("lib"))
```
Comment (waruqi) [inline packages/c/clip/xmake.lua:33]: please move dll to bin directory

File: packages/c/clip/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    end)
+
+    on_test(function(package)
+        assert(package:check_cxxsnippets( {test = [[
+              #include <clip/clip.h>
+              #include <string>
+              void test() { clip::set_text("foo"); }
+          ]]},
+          { configs = "" }
```
Comment (waruqi) [inline packages/c/clip/xmake.lua:42]: empty config? please remove it

**Comment by waruqi** [comment]
> It's building everywhere but Windows shared targets. They are failing on an unresolved external from `check_cxxsnippets` test:
> 
> ```shell
> error LNK2019: unresolved external symbol "bool __cdecl clip::set_text(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" 
> ```
> 
> I tried moving the dll from `bin` to `lib` but same result.

maybe you need export some symbols for dll.

File: packages/c/clip/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    add_includedirs("include")
+
+    on_install("linux", "windows", "mingw@windows,msys", "macos", "bsd", function(package)
+        local configs = {}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DCLIP_EXAMPLES=OFF")
+        table.insert(configs, "-DCLIP_TESTS=OFF")
+        if package:config("shared") and is_plat("windows") then
```
Comment (waruqi) [inline packages/c/clip/xmake.lua:27]: use package:is_plat

**Pull request merged**